armory/instrument/instrument.py

```diff
@@ -0,0 +1,690 @@
+"""
+OOP structure for Armory logging
+
+Example 1:
+    Use case - measure L2 distance of post-preprocessing for benign and adversarial
+    Code:
+        # in model
+        from armory import instrument
+        probe = instrument.get_probe("model")
+        ...
+        x_post = model_preprocessor(x)
+        probe.update(lambda x: x.detach().cpu().numpy(), x_post=x_post)
+
+        # outside of model code
+        probe.hook(model, lambda x: x.detach().cpu().numpy(), x_post=x_post)
+
+        # elsewhere (could be reasonably defined in a config file as well)
+        from armory import instrument
+        from armory import metrics
+        meter = instrument.MetricMeter("l2_dist_postprocess", metrics.L2, "model.x_post[benign]", "model.x_post[adversarial]")
+        instrument.add_meter(meter)
```

Comment:
not sure if these comments are meant to make it into the merge, but if so, the instrument module no longer has a `add_meter` function